src/useDeviceOrientation.tsx

```diff
@@ -1,16 +1,19 @@
-import { useEffect, useState } from 'react'
-import { useWindowDimensions } from 'react-native'
+import { useEffect, useState } from 'react';
+import { useWindowDimensions } from 'react-native';
 
-function useDeviceOrientation() {
-   const { height, width } = useWindowDimensions()
+// TODO: Comment
+export function useDeviceOrientation() {
+  const { height, width } = useWindowDimensions();
 
-   const [isPortrait, setIsPortrait] = useState<boolean>(height >= width)
+  const [isPortrait, setIsPortrait] = useState<boolean>(height >= width);
 
-   useEffect(() => {
-      setIsPortrait(height >= width)
-   }, [width])
+  useEffect(() => {
+    setIsPortrait(height >= width);
+    //  TODO: why is height not included in the dependency array?
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [width]);
```

Comment:
This is because the use effect gets triggered when the width changes.

Reply:
But shouldn't it also be triggered when the height changes?  It's a long explanation, but react holds onto the previous value of height unless it's put into the dependencies, so this can cause some funky bugs.